modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -129,6 +132,12 @@ public void processOpts() {
         if (additionalProperties.containsKey("localVariablePrefix")) {
             this.setLocalVariablePrefix((String) additionalProperties.get("localVariablePrefix"));
         }
+        
+        if (additionalProperties.containsKey("serializableModel")) {
+            this.setSerializableModel(Boolean.valueOf((String)additionalProperties.get("serializableModel")));
```

Comment:
@fehguy I tested again with the `serializable_java_models` branch but couldn't repeat the `ClassCastException`. (test cases: serializableModel no set, serializableModel set to true, serializableModel set to false)

I discovered another issue that `Serializable` is used even though `serializableModel` is set to false (previously I tested with true and no value set). I'll submit a PR to address it.